### ISSUE TITLE
eth_getTransactionByHash returns INVALID_PARAMS for unmined tx

### ISSUE THREAD

**Issue by ethers** [ethereum/cpp-ethereum#2173]
develop deb0f46

eth returns INVALID_PARAMS:
```
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x2df7c9a446f6e4729d47a80acaf0f844190d40aa87ee37d620bc3c3fc974a4b3"],"id":1}' http://localhost:8549 | jq '.'
{
  "jsonrpc": "2.0",
  "id": 1,
  "error": {
    "message": "INVALID_PARAMS: Invalid method parameters (invalid name and/or type) recognised",
    "code": -32602
  }
}
```

Here's what Geth returns:
```
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x2df7c9a446f6e4729d47a80acaf0f844190d40aa87ee37d620bc3c3fc974a4b3"],"id":1}' http://localhost:8545 | jq '.'
{
  "result": null,
  "jsonrpc": "2.0",
  "id": 1
}
```

**Comment by debris** [comment]
fixed in #2183

**Comment by debris** [comment]
merged in #2343
